test/hotspot/jtreg/compiler/vectorapi/VectorRearrangeTest.java

```diff
@@ -0,0 +1,222 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8350463
+ * @summary AArch64: Add vector rearrange support for small lane count vectors
+ * @requires (os.simpleArch == "x64" & vm.cpu.features ~= ".*avx.*") | os.arch=="aarch64"
+ * @modules jdk.incubator.vector
+ * @library /test/lib /
+ *
+ * @run driver compiler.vectorapi.VectorRearrangeTest
+ */
+
+package compiler.vectorapi;
+
+import jdk.incubator.vector.*;
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+public class VectorRearrangeTest {
+    private static final int LENGTH = 2048;
+    private static final Random random = Utils.getRandomInstance();
+
+    private static final VectorSpecies<Byte> bspec128    = ByteVector.SPECIES_128;
+    private static final VectorSpecies<Short> sspec128   = ShortVector.SPECIES_128;
+    private static final VectorSpecies<Integer> ispec128 = IntVector.SPECIES_128;
+    private static final VectorSpecies<Long> lspec128    = LongVector.SPECIES_128;
+    private static final VectorSpecies<Float> fspec128   = FloatVector.SPECIES_128;
+    private static final VectorSpecies<Double> dspec128  = DoubleVector.SPECIES_128;
+    private static final VectorSpecies<Byte> bspec64     = ByteVector.SPECIES_64;
+    private static final VectorSpecies<Short> sspec64    = ShortVector.SPECIES_64;
+    private static final VectorSpecies<Integer> ispec64  = IntVector.SPECIES_64;
+    private static final VectorSpecies<Float> fspec64    = FloatVector.SPECIES_64;
+
+    private static byte[]   bsrc;
+    private static short[]  ssrc;
+    private static int[]    isrc;
+    private static long[]   lsrc;
+    private static float[]  fsrc;
+    private static double[] dsrc;
+
+    private static byte[]   bdst;
+    private static short[]  sdst;
+    private static int[]    idst;
+    private static long[]   ldst;
+    private static float[]  fdst;
+    private static double[] ddst;
+
+    private static int[][] indexes;
+
+    static {
+        bsrc = new byte[LENGTH];
+        ssrc = new short[LENGTH];
+        isrc = new int[LENGTH];
+        lsrc = new long[LENGTH];
+        fsrc = new float[LENGTH];
+        dsrc = new double[LENGTH];
+        bdst = new byte[LENGTH];
+        sdst = new short[LENGTH];
+        idst = new int[LENGTH];
+        ldst = new long[LENGTH];
+        fdst = new float[LENGTH];
+        ddst = new double[LENGTH];
+
+        for (int i = 0; i < LENGTH; ++i) {
+            bsrc[i] = (byte)random.nextInt();
+            ssrc[i] = (short)random.nextInt();
+            isrc[i] = random.nextInt();
+            lsrc[i] = random.nextLong();
+            fsrc[i] = random.nextFloat();
+            dsrc[i] = random.nextDouble();
+        }
```

Comment:
Could you please use `Generators.java`? This makes sure that we have more "interesting" values in the distribution.

Reply:
Thanks for your suggestion! I will take a look at this interface.